test/e2e/util/e2e_determined_client.go

```diff
@@ -0,0 +1,54 @@
+package util
+
+import (
+	"context"
+
+	k8scontrollerclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/onsi/gomega"
+)
+
+// DeterminedE2EClient wraps E2eClient calls in an Eventually assertion to keep trying
+// or bail after some time if unsuccessful
+type DeterminedE2EClient struct {
+	*E2EKubeClient
+}
+
+func NewDeterminedClient(e2eKubeClient *E2EKubeClient) *DeterminedE2EClient {
+	return &DeterminedE2EClient{
+		e2eKubeClient,
+	}
+}
+
+func (m *DeterminedE2EClient) Create(context context.Context, obj k8scontrollerclient.Object, options ...k8scontrollerclient.CreateOption) error {
+	m.keepTrying(func() error {
+		err := m.E2EKubeClient.Create(context, obj, options...)
+		return err
+	})
+	return nil
+}
+
+func (m *DeterminedE2EClient) Update(context context.Context, obj k8scontrollerclient.Object, options ...k8scontrollerclient.UpdateOption) error {
+	m.keepTrying(func() error {
```

Comment:
should we check for conflict failures and get the latest resource version before trying again?